flink-ml-lib/src/main/java/org/apache/flink/ml/feature/standardscaler/StandardScalerParams.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.feature.standardscaler;
+
+import org.apache.flink.ml.common.param.HasFeaturesCol;
+import org.apache.flink.ml.common.param.HasPredictionCol;
+import org.apache.flink.ml.param.BooleanParam;
+import org.apache.flink.ml.param.Param;
+
+/**
+ * Params for {@link StandardScaler}.
+ *
+ * @param <T> The class type of this instance.
+ */
+public interface StandardScalerParams<T> extends HasFeaturesCol<T>, HasPredictionCol<T> {
```

Comment:
Spark's `StandardScaler` uses `inputCol` and `outputCol` as its parameters. Would it be better to follow the same approach here?

Note that this `StandardScaler` is used to transform features rather than making predictions. Thus it does not seem intuitive to name its output as `predictionCol`. We can submit a followup PR to fix this problem.

`HasFeaturesCol` should probably be renamed to `HasFeatureCol` as it represents just one column.

And `HasInputCol`  is more consistent with `HasOutputCol` than `HasFeatureCol` does.

Reply:
Thanks for raising up this discussion. 

As `StandardScaler` is an `estmator`, it aims to train on the input features and predict on the output. So I thought it might be ok to keep the current naming. So soes `MinMaxScaler`.

> HasFeaturesCol should probably be renamed to HasFeatureCol as it represents just one column.
Here `features` is one column.

Reply:
After some offline discussion, we agree to follow the following rules:
- use `inputCol/outputCol/inputCols/outputCos` when doing feature engineering. (e.g., code under package `org.apache.flink.ml.feature`)
- use `featuresCol/predictionCol` when doing (un)supervised machine learning training (e.g., code under package `org.apache.flink.ml.classification, org.apache.flink.ml.clustering`